src/tests/tracing/eventpipe/common/IpcEndpointConfig.cs

```diff
@@ -0,0 +1,144 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Tracing.Tests.Common
+{
+    internal class IpcEndpointConfig
+    {
+        public enum TransportType
+        {
+            NamedPipe,
+            UnixDomainSocket,
+            TcpClient
+        }
+
+        TransportType _transportType;
+
+        public string Address { get; }
+
+        public TransportType Transport => _transportType;
+
+        const string NamedPipeSchema = "namedpipe";
+        const string UnixDomainSocketSchema = "uds";
+        const string NamedPipeDefaultIPCRoot = @"\\.\pipe\";
+        const string NamedPipeSchemaDefaultIPCRootPath = "/pipe/";
+
+        public IpcEndpointConfig(string address, TransportType transportType)
+        {
+            if (string.IsNullOrEmpty(address))
+                throw new ArgumentException("Address is null or empty.");
+
+            switch (transportType)
+            {
+                case TransportType.NamedPipe:
+                {
+                    if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                        throw new PlatformNotSupportedException($"{NamedPipeSchema} is only supported on Windows.");
+                    break;
+                }
+                case TransportType.UnixDomainSocket:
+                {
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                        throw new PlatformNotSupportedException($"{UnixDomainSocketSchema} is not supported on Windows, use {NamedPipeSchema}.");
+                    break;
+                }
+                case TransportType.TcpClient:
+                {
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                        throw new PlatformNotSupportedException($"TcpClient is not supported on Windows, use {NamedPipeSchema}."); // It is probably supported, just not recommended
```

Comment:
Maybe we shouldn't throw if its windows (I'm guessing it works on windows), but instead check if its Android, and if not throw with expectation that TcpClient is used on Android